### PR TITLE
Add mousedown drag to build a relation

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Drop to services on the canvas that can be related
- Click the relation icon on a service and click the other service to check the old method still works
- Add another service that can relate to one of the services
- Click and drag the relation icon of the new service to the relate-able service, mouse up and check the relation is made
- To QA the "Stop a relation line starting a new one while dragging" bug fix. I would suggest its easier to checkout `2e76e5f` which is the commit before the fix and drag the line around. You should see that some relation lines get left behind
- Then check out `a9b8382` with the fix and do the same. You should not see any orphan relation lines left behind

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5552/
Test PASSed.

File: jujugui/static/gui/src/app/views/topology/relation.js [hunk 1]
```diff
@@ -545,29 +545,32 @@ YUI.add('juju-topology-relation', function(Y) {
...
-      var point = self.cursorBox.getConnectorPair(d);
-      dragline.attr('x1', point[0][0])
-              .attr('y1', point[0][1])
-              .attr('x2', point[1][0])
-              .attr('y2', point[1][1]);
-      self.dragline = dragline;
-      vis.select('.plus-service').classed('fade', true);
-      // Start the add-relation process.
-      self.addRelationStart(d, self);
+      // Only start a new drag line if not already dragging
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/relation.js]: Can you add to this comment the answer to "Why would it be?"

**Comment by hatched** [comment]
Can this not be tested?

**Comment by hatched** [comment]
Thanks @anthonydillon :+1: :ok: 

Would be nice to get a test here though.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5560/
Test PASSed.

File: jujugui/static/gui/src/app/views/topology/relation.js [hunk 2]
```diff
@@ -545,29 +545,33 @@ YUI.add('juju-topology-relation', function(Y) {
...
+        // Create a pending drag-line.
+        var vis = this.get('component').vis;
+        var dragline = vis.insert('line',':first-child')
+                          .attr('class',
+                                'relation pending-relation dragline dragging');
+        var self = this;
+
+        // Start the line between the cursor and the nearest connector
+        // point on the service.
+        this.set('dragplane', Y.one('.the-canvas g').getDOMNode());
```
Comment (makyo) [inline jujugui/static/gui/src/app/views/topology/relation.js]: `vis.select(...)`? *shrug*

**Comment by makyo** [comment]
👍 Looks good

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5564/
Test PASSed.

**Comment by anthonydillon** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**